decoder.js

```diff
@@ -136,8 +138,18 @@ GIFDecoder.prototype._readPalette = function() {
     
   this._palette = buf.slice(0, len);
   buf.consume(len);
-  
-  this._state = this._state === GIF_GLOBAL_PALETTE ? GIF_BLOCK : GIF_LZW;
+
+  if (this._state === GIF_GLOBAL_PALETTE) {
+    if (this._outputIndexed) {
+      this.format.palette = this._palette.slice(0);
```

Comment:
I would not do a copy, just for performance reasons. Most of the time users won't modify the palette, and if they do it's their problem to make a copy.